src/azure-cli/azure/cli/command_modules/appservice/_validators.py

```diff
@@ -242,7 +243,10 @@ def validate_add_vnet(cmd, namespace):
     webapp_loc = _normalize_location(cmd, webapp.location)
     vnet_loc = _normalize_location(cmd, vnet_loc)
 
-    if vnet_loc != webapp_loc:
+    nonstage_webapp_loc = _normalize_stage_location(webapp_loc)
+    nonstage_vnet_loc = _normalize_stage_location(vnet_loc)
+
+    if nonstage_vnet_loc != nonstage_webapp_loc:
         raise ValidationError("The app and the vnet resources are in different locations. "
```

Comment:
include the actual locations that were compared in the error message.  In this case nonstage_vnet_loc and nonstage_webapp_loc